src/client/theme-default/components/VPSidebarItem.vue

```diff
@@ -171,36 +185,30 @@ function onCaretClick() {
   color: var(--vp-c-text-2);
 }
 
-.VPSidebarItem.level-0.is-link > .item > .link:hover .text,
-.VPSidebarItem.level-1.is-link > .item > .link:hover .text,
-.VPSidebarItem.level-2.is-link > .item > .link:hover .text,
-.VPSidebarItem.level-3.is-link > .item > .link:hover .text,
-.VPSidebarItem.level-4.is-link > .item > .link:hover .text,
-.VPSidebarItem.level-5.is-link > .item > .link:hover .text {
-  color: var(--vp-c-brand-1);
-}
-
-.VPSidebarItem.level-0.has-active > .item > .text,
-.VPSidebarItem.level-1.has-active > .item > .text,
-.VPSidebarItem.level-2.has-active > .item > .text,
-.VPSidebarItem.level-3.has-active > .item > .text,
-.VPSidebarItem.level-4.has-active > .item > .text,
-.VPSidebarItem.level-5.has-active > .item > .text,
-.VPSidebarItem.level-0.has-active > .item > .link > .text,
-.VPSidebarItem.level-1.has-active > .item > .link > .text,
-.VPSidebarItem.level-2.has-active > .item > .link > .text,
-.VPSidebarItem.level-3.has-active > .item > .link > .text,
-.VPSidebarItem.level-4.has-active > .item > .link > .text,
-.VPSidebarItem.level-5.has-active > .item > .link > .text {
-  color: var(--vp-c-text-1);
-}
-
-.VPSidebarItem.level-0.is-active > .item .link > .text,
-.VPSidebarItem.level-1.is-active > .item .link > .text,
-.VPSidebarItem.level-2.is-active > .item .link > .text,
-.VPSidebarItem.level-3.is-active > .item .link > .text,
-.VPSidebarItem.level-4.is-active > .item .link > .text,
-.VPSidebarItem.level-5.is-active > .item .link > .text {
+.VPSidebarItem.level-0.is-active > .item > summary .text,
+.VPSidebarItem.level-1.is-active > .item > summary .text,
+.VPSidebarItem.level-2.is-active > .item > summary .text,
+.VPSidebarItem.level-3.is-active > .item > summary .text,
+.VPSidebarItem.level-4.is-active > .item > summary .text,
+.VPSidebarItem.level-5.is-active > .item > summary .text,
+.VPSidebarItem.level-0.is-active > .item > .link > .text,
+.VPSidebarItem.level-1.is-active > .item > .link > .text,
+.VPSidebarItem.level-2.is-active > .item > .link > .text,
+.VPSidebarItem.level-3.is-active > .item > .link > .text,
+.VPSidebarItem.level-4.is-active > .item > .link > .text,
+.VPSidebarItem.level-5.is-active > .item > .link > .text,
+.VPSidebarItem.level-0 .link:hover >.text,
+.VPSidebarItem.level-1 .link:hover >.text,
+.VPSidebarItem.level-2 .link:hover >.text,
+.VPSidebarItem.level-3 .link:hover >.text,
+.VPSidebarItem.level-4 .link:hover >.text,
+.VPSidebarItem.level-5 .link:hover >.text,
+.VPSidebarItem.level-0 .link:focus >.text,
+.VPSidebarItem.level-1 .link:focus >.text,
+.VPSidebarItem.level-2 .link:focus >.text,
+.VPSidebarItem.level-3 .link:focus >.text,
+.VPSidebarItem.level-4 .link:focus >.text,
+.VPSidebarItem.level-5 .link:focus >.text {
```

Comment:
In a vacuum, I'd write this as

```css
.VPSidebarItem:where(.level-0, .level-1, .level-2, .level-3, .level-4, .level-5) {
  &.is-active > .item {
    > summary .text {
      color: var(--vp-c-brand-1);
    }

    > .link > .text {
      color: var(--vp-c-brand-1);
    }
  }

  .link:where(:hover,:focus) .text {
    color: var(--vp-c-brand-1);
  }
}
```

That would survive the build process, so a refactor would reduce weight (1244 -> 311 for this block). But I don't find any uses of native CSS nesting ([caniuse](https://caniuse.com/css-nesting)) in this codebase, let alone this kind of `:where()` ([caniuse](https://caniuse.com/mdn-css_selectors_where)) use.

Lmk if you'd like that refactor. Could roll it into this PR. Could refactor some of this components' existing CSS to get more reduction. Could open a new issue for a more widespread refactor. Or could leave it alone!

Reply:
CSS nesting does not align with our baseline. `:where` does. We are using `:is` at some places and their compatibility is similar. You can use that. Don't use `:has` though.

Reply:
Updated

//

Huh look at that GitHub fanciness. It's prefixing linked refs in commit messages with `vuejs`. Hadn't noticed that before.

//

Thanks for the info! Is the browser support documented? I haven't found it.

Reply:
It's same as Vite - https://vitejs.dev/guide/build.html#browser-compatibility (except for chrome v87 which supports ESM but doesn't support `:is`)